go/consensus/api/api.go

```diff
@@ -0,0 +1,153 @@
+// Package consensus provides the implementation agnostic consensus
+// backend.
+package api
+
+import (
+	"context"
+
+	beacon "github.com/oasislabs/oasis-core/go/beacon/api"
+	"github.com/oasislabs/oasis-core/go/common/crypto/signature"
+	"github.com/oasislabs/oasis-core/go/common/node"
+	"github.com/oasislabs/oasis-core/go/consensus/api/transaction"
+	epochtime "github.com/oasislabs/oasis-core/go/epochtime/api"
+	genesisAPI "github.com/oasislabs/oasis-core/go/genesis/api"
+	keymanager "github.com/oasislabs/oasis-core/go/keymanager/api"
+	registry "github.com/oasislabs/oasis-core/go/registry/api"
+	roothash "github.com/oasislabs/oasis-core/go/roothash/api"
+	scheduler "github.com/oasislabs/oasis-core/go/scheduler/api"
+	staking "github.com/oasislabs/oasis-core/go/staking/api"
+)
+
+// ClientBackend is a limited consensus interface used by clients that
+// connect to the local node.
+type ClientBackend interface {
+	// SubmitTx submits a signed consensus transaction.
+	SubmitTx(ctx context.Context, tx *transaction.SignedTransaction) error
+
+	// TODO: Add things like following consensus blocks.
```

Comment:
will this be done in this PR? (otherwise open an issue)

Reply:
No it will not be done as it has nothing to do with the transaction format :-) I'll open an issue, this is something that the blockexploder people will need.